src/Control/Effect/NonDet.hs

```diff
@@ -1,4 +1,16 @@
 {-# LANGUAGE TypeOperators #-}
+
+{- | An effect modelling nondeterminism with choice and failure.
+
+Nondeterministic operations are encapsulated by the 'Alternative' class, where 'empty' represents failure and '<|>' represents choice. This module reexports the 'Alternative' interface. If you can't or don't want to use 'Alternative', you can use the 'Control.Effect.Empty.empty' and 'Control.Effect.Choose.<|>' effects (from 'Control.Effect.Empty' and 'Control.Effect.Choose') directly, as the 'NonDet' effect is the composition of 'Choose' and 'Empty'.
+
+Predefined carriers:
+
+* "Control.Carrier.NonDet.Church", which collects all branches' results using an @Alternative@ functor.
+* "Control.Carrier.NonDet.Maybe", which terminates upon encountering the first successful result.
```

Comment:
```suggestion
* "Control.Carrier.NonDet.Maybe", which returns at most one result, in `Maybe`.
```

(It’s kind of a big deal that `NonDet.Church` can deal with infinite search spaces, so I don’t want to give the impression that `.Maybe` is in some way special. It’s sometimes more convenient, that’s all.)